added default working directory when no working directory is specified

JohnTGZ opened the pull request:
<!--

Thank you for your contribution!

Get early feedback through
- Dronecode Discord: https://discord.gg/dronecode
- PX4 Discuss: http://discuss.px4.io/
- opening a draft pr and sharing the link

-->

### Solved Problem
This fixes #22419, where the PX4-SITL instance dies due to wrong default working directory.  

### Solution
- Added default working directory when no working directory is specified to the arguments for the `px4` executable

### Alternatives
As a temporary fix without modifying the source code, I modified the [px4 node in px4.launch](https://github.com/PX4/PX4-Autopilot/blob/c7672fc68c334121f8a41141a137b33a68da6bd4/launch/px4.launch#L15C76-L15C76) to specify the working directory path explicitly, as shown below:
```
    <node name="sitl_$(arg ID)" pkg="px4" type="px4" output="screen" args="$(find px4)/build/px4_sitl_default/etc -s etc/init.d-posix/rcS -i $(arg ID) $(arg px4_command_arg1) -w $(find px4)/build/px4_sitl_default/rootfs">
```

JohnTGZ committed:
added default working directory when no working directory is specified

dagar commented on platforms/posix/src/px4/common/main.cpp:
```diff
@@ -286,6 +286,10 @@ int main(int argc, char **argv)
 			commands_file = "etc/init.d-posix/rcS";
 		}
 
+		if (working_directory.empty()){
```
```suggestion
		if (working_directory.empty()) {
```

dagar replied:
Check if the path exists.

JohnTGZ replied:
Alright, I decided to only set a default working directory with the `PX4_SOURCE_DIR` only if `PX4_SOURCE_DIR` is defined, and also added checking of the `working_directory` path

JohnTGZ committed:
added checking of path and initialized default default working_directory only if PX4_SOURCE_DIR is defined

JohnTGZ committed:
correction to working directory path check

dagar commented:
CI SITL tests are failing, need to investigate.

JohnTGZ committed:
Moved working_directory check block

Ensure that this executes regardless of whether PX4_SOURCE_DIR is defined or not
